.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,22 @@
+name: Integration Tests
+on:
+  # Run tests on main just so the module and build cache can be saved and used
+  # in PRs. This speeds up the time it takes to test PRs dramatically.
```

Comment:
So the build cache of main is used when building PRs, but the build cache of one PR is not used when building other PRs? I thought that the cache would be something stored on a build server, and it'd be used regardless of which branch is being built.

Reply:
"I thought that the cache would be something stored on a build server, and it'd be used regardless of which branch is being built." -> I believe that this is correct. I think that the point is to always have a fresh cache available from the main branch that PRs can use when building for the first time

Reply:
When is this workflow ran on the `main` branch? Is it when a PR is merged to it? If the test already ran during the PR, I'm not sure if there is any point in having them also run after the PR is merged? I guess the cache would be the same.

Reply:
I agree but @rfratto removed it and they put it back for the unit tests in https://github.com/grafana/agent/pull/2921. Robert in the PR you say that "it turns out commits to main are responsible for storing the module and build cache", but aren't PRs also building the same cache?

Reply:
Hmm ok. I am not sure why this is, but if you could clarify in the comment in `.github/workflows/integration-tests.yml` that the cache will only be populated if we run this on `main` and not just the CI of any other branch? It'd be nice if @rfratto could help clarify further why this is, if he happens to remember the reason.

Reply:
I think we cant try without it and see how it behaves. If we see that the PRs don't hit the cache we can investigate further

Reply:
> If we see that the PRs don't hit the cache

Do you know how to check for this? IS there a log line?

Reply:
If the tests run under 4 mins then the cache was hit, if it runs in around 14mins that means that it spent 10mins to download the dependencies instead of using the cache